pkg/api/graph_test.go

```diff
@@ -143,6 +143,18 @@ func TestBuildGraph(t *testing.T) {
 		creates:  []StepLink{InternalImageLink(PipelineImageStreamTagReference("final"))},
 	}
 
+	duplicateRoot := &fakeStep{
+		requires: []StepLink{ExternalImageLink(ImageStreamTagReference{Namespace: "ns", Name: "base", Tag: "latest"})},
+		creates:  []StepLink{InternalImageLink(PipelineImageStreamTagReferenceRoot)},
+	}
+	duplicateSrc := &fakeStep{
+		requires: []StepLink{
+			InternalImageLink(PipelineImageStreamTagReferenceRoot),
```

Comment:
Oh ew we should just disallow this config?

Reply:
Provides() is a dynamic dependency.  We could make the step links resolve themselves, but it's a pretty obvious error to make.